src/marks/text.js

```diff
@@ -406,7 +406,9 @@ function defaultWidth(text, start, end) {
 }
 
 function monospaceWidth(text, start, end) {
-  return 100 * (end - start);
+  let sum = end - start;
+  for (let i = start; i < end; ++i) sum -= isSurrogatePair(text[i], text[i + 1]);
```

Comment:
If a surrogate pair starts at _i_ (_i.e._, [_i_, _i_ + 1] is a surrogate pair) then we shouldn’t test whether another surrogate pair starts at _i_ + 1 (_i.e._, [_i_ + 1, _i_ + 2]); we should skip an index. (Hence the `i += …` in `defaultWidth`.)

Reply:
oh but monospace emoji _are_ very much larger than the other chars! I'm sending what I have, and will review tomorrow with fresh eyes.

Reply:
We shouldn’t assume that all surrogate pairs are emojis. If you want to add testing specifically for emojis, I suppose we could do that (testing for the [Unicode emoticons block](https://en.wikipedia.org/wiki/Emoticons_(Unicode_block))), but my thinking is that we shouldn’t assume anything about the characters, at least for the monospace case—we should interpret monospace as “all characters have equal width” even if they don’t in practice.